HorizonEditor/src/ContentBrowser/ContentBrowser.cpp

```diff
@@ -0,0 +1,148 @@
+#include "imgui_internal.h"
+#include "EditorLayer.h"
+#include "Modals.h"
+
+void ContentBrowser::OnImGuiRender()
+{
+
+
+	ImGui::Begin("Content Browser");
+
+	if (!Modals::projectRootFolder.empty()) {
+		if (Modals::m_CurrentDirectory != std::filesystem::path(m_ProjectRootFolder))
+		{
+			if (ImGui::Button("<-"))
+			{
+				Modals::m_CurrentDirectory = Modals::m_CurrentDirectory.parent_path();
+			}
+		}
+
+		static float padding = 16.0f;
+		static float thumbnailSize = 128.0f;
+		float cellSize = thumbnailSize + padding;
+
+		float panelWidth = ImGui::GetContentRegionAvail().x;
+		int columnCount = (int)(panelWidth / cellSize);
+		if (columnCount < 1)
+			columnCount = 1;
+
+		ImGui::Columns(columnCount, 0, false);
+
+
+
+		for (auto& entry : std::filesystem::directory_iterator(Modals::m_CurrentDirectory))
+		{
+
+			if (entry.path().extension().string().find("meta") != std::string::npos || entry.path().extension().string().find("ini") != std::string::npos || entry.path().extension().string().find("hzn") != std::string::npos || entry.path().string().find("icons") != std::string::npos)
+			{
+				continue;
+			}
+
+			const auto& path = entry.path();
+			std::string filenameString = path.filename().string();
+
+			ImGui::PushID(filenameString.c_str());
+
+
+			std::shared_ptr<Hzn::Texture> icon;
+
+			if (entry.is_directory()) {
+				icon = Modals::folderIcon;
+			}
+
+			else if (entry.path().string().find(".png") != std::string::npos)
+			{
+				icon = Modals::assetManager.GetTexture(entry.path().string());
+			}
+
+			else
+			{
+				icon = Modals::fileIcon;
+			}
+			ImGui::PushStyleColor(ImGuiCol_Button, ImVec4(0, 0, 0, 0));
+
+			if (entry.path().parent_path().string().find("sprites") != std::string::npos)
+			{
+
+
+				for (const auto& metaFile : std::filesystem::recursive_directory_iterator(entry.path().parent_path())) {
+
+					if (metaFile.path().string().find(".meta") != std::string::npos && metaFile.path().filename().string().substr(0, metaFile.path().filename().string().find(".")) == entry.path().filename().string().substr(0, entry.path().filename().string().find("."))) {
+						std::ifstream infile(metaFile.path().c_str(), std::ifstream::binary);
+						std::string line;
+
+						while (std::getline(infile, line)) {
+							std::istringstream is_line(line);
+							std::string key;
+							if (std::getline(is_line, key, ':'))
+							{
+								std::string value;
+
+								if (std::getline(is_line, value))
+								{
+									Modals::spriteFormat[key] = value;
+								}
+							}
+						}
+
+					}
+				}
+
+				ImGui::ImageButton((ImTextureID)icon->getId(), { thumbnailSize, thumbnailSize }, { 0, 1 }, { 1, 0 });
+
+
+				if (ImGui::Button("show sprites")) {
+					m_CurrentTexturePath = entry.path().string();
+				}
+
+			}
+			else {
+				ImGui::ImageButton((ImTextureID)icon->getId(), { thumbnailSize, thumbnailSize }, { 0, 1 }, { 1, 0 });
+			}
+
+			ImGui::PopStyleColor();
+
+			if (ImGui::BeginDragDropSource()) {
```

Comment:
Is this the section that incorporates the drag and drop code for the content browser?

Reply:
yes.Drag and drop consists of two parts: drag and drop source and drag and drop target